### ISSUE TITLE
multiprocessing function cannot use when calling from a DDP process

### ISSUE THREAD

**Issue by shamanez** [pytorch/pytorch#55156]
## 🐛 Bug

I am using PyTorch-lightning to train my model with multiple GPUs. During the training, I want to run another parallel process that updates some embeddings by sending them through a part of the network and saves them into the hard disk.  This part has nothing to do with gradient updates. From my main training process, I have to pass the part of my model to the multiprocessing function. But I can't pass any  'cuda' objects. Please refer to the following code.

```
    def training_step(self, batch, batch_idx) -> Dict:

    
        global stepCount
        global isOtherThreadBusy
        global threadHandle


        if (self.trainer.global_rank==0):

                if not isOtherThreadBusy:
         
                    ctx_encoder=self.trainer.model.module.module.model.rag.ctx_encoder
                    model_copy =type(ctx_encoder)(self.config_dpr) # get a new instance  #this will be load in the CPU
                    model_copy.load_state_dict(ctx_encoder.state_dict()) # copy weights and stuff

                    threadHandle =Process(target=self.indexed_function,args=(model_copy,))
                    threadHandle.start()
                    isOtherThreadBusy = True



```

When the model_copy being attached to a cuda_device I get errors. I tried  adding     # **multiprocessing.set_start_method('spawn')**. But no luck.
